### PR TITLE
/sheets-data endpoint

### LINKED ISSUE #21: Index data from Mongo DB
Source data (if you want it): https://docs.google.com/spreadsheets/d/18ZWXfEeGEP8NV5g_flmEhBkXgsKEJT6y9iHt0XZ_2lE/edit#gid=0

- add npm package to read from mongo and save to es
- get the form data with an endpoint
- pass that to {new} and {save} will save it to mongo db
- the other tool will pass from mongo to es

### PR DESCRIPTION
**Pull request by cy**
closes #21 
endpoint to read goggle sheets data. todo: better way to generate/store token, in .env for now

GET http://localhost:5050/sheets-data
```
[
{
"Model Name": "HCM-BROD-9989.C79",
"Model Type": "3-D: Other (neurosphere)",
"Growth Rate": "",
"Split Ratio": "",
"Gender": "Female",
"Race": "White",
"Age at Diagnosis": "50",
"Age at Sample Acquisition": "56",
"Date of Availability": "",
"Sequence Source": "",
"Source Model UUID": "",
"Source Sequence UUID": "",
"Primary Site": "Brain",
"pT Stage*": "pT2",
"pN Stage": "pN1c",
"pM Stage": "pM1",
"Neoadjuvant Therapy": "Yes, both radiation and pharmaceutical treatment",
"Chemotherapeutic Drugs": "No",
"Disease Status": "Progression",
"Vital Status": "Alive",
"Therapy": "Unknown",
"Molecular Characterizations": "",
"Clinical Tumor Diagnosis": "Breast cancer",
"Histological Type": "Metaplastic ductal carcinoma",
"Clinical Stage Grouping": "",
"Site of Sample Acquisition": "Brain",
"Tumor Histological Grade": "3"
}, [etc]
]
```

### CODE REVIEW

**Comment by cy** [comment]
cleaning this up and adding a bin script to getToken

**Comment by cy** [comment]
@ra-ver cleaned up and ready for review, DM on slack for my .env or create client_id in google app console.

To generate a token, run `node bin/getToken.js` from `/api`, which will output a url. Go to the url, login with google, which will redirect to`http://localhost:5050/?code=`, paste that code into console and a token.json will be saved.

File: api/bin/getToken.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      oAuth2Client.setCredentials(token);
+      // Store the token to disk for later program executions
+      fs.writeFile(TOKEN_PATH, JSON.stringify(token), (err) => {
+        if (err) console.error(err);
+        console.log('Token stored to', TOKEN_PATH);
+      });
+      callback(oAuth2Client);
+    });
+  });
+}
```
Comment (ra-ver) [inline api/bin/getToken.js]: this is really nice 💯

File: api/package.json [hunk 2]
```diff
@@ -15,7 +15,10 @@
     "babel-polyfill": "^6.26.0",
     "dotenv": "^5.0.1",
     "elasticsearch": "^14.0.0",
-    "express-inspector": "^0.3.1"
+    "express-inspector": "^0.3.1",
+    "fs": "^0.0.1-security",
+    "googleapis": "27",
+    "mongoose": "^5.1.3"
```
Comment (ra-ver) [inline api/package.json]: let's not mix the search api with CMS api..could we refactor these things to another folder called "cms" or "cms-api" or anything else that seems more suitable? so the repo will have 4 root folders:
/api
/cms
/data_model
/ui

File: api/src/SheetsToMongo.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    return (
+      {
+        yes: true,
+        no: false,
+        true: true,
+        false: false,
+      }[(data || '').toLowerCase()] || false
+    );
+  },
+};
```
Comment (ra-ver) [inline api/src/SheetsToMongo.js]: lets move this to /cms .. see comment above.

File: api/src/index.js [hunk 4]
```diff
@@ -14,6 +20,43 @@ app.use(cors());
...
+          return model.save();
+        });
+
+        Promise.all(savePromises)
+          .then(docs => res.json({ docs }))
+          .catch(error => error);
+      })
+      .catch(error => res.json({ error: `error reading sheet, ${error}` }));
+  });
+
```
Comment (ra-ver) [inline api/src/index.js]: lets move this to /cms as well

File: api/src/schemas/model.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  clinical_tumor_diagnosis: String,
+  histological_type: String,
+  clinical_stage_grouping: String,
+  site_of_sample_acquisition: String,
+  tumor_histological_grade: Number,
+  histopathological_biomarkers: String,
+  requirements_for_3rd_party_licensing: String,
+});
+
+export default mongoose.model('Model', ModelSchema);
```
Comment (ra-ver) [inline api/src/schemas/model.js]: to /cms as well

File: api/bin/getToken.js (hunk 1 above)
Comment (cy) [reply api/bin/getToken.js]: mostly from google sheets api guide :sweat_smile:

**Pull request merged**